optuna/storages/_cached_storage.py

```diff
@@ -328,7 +333,7 @@ def get_all_trials(
         states: Optional[Container[TrialState]] = None,
     ) -> List[FrozenTrial]:
         if study_id not in self._studies:
-            self.read_trials_from_remote_storage(study_id)
+            self.read_trials_from_remote_storage(study_id, sync_owned_trials=False)
```

Comment:
Why can we skip syncing owned trials here? If external processes can access & update them, I think we should always sync them.